format/FlightSql.proto

```diff
@@ -0,0 +1,454 @@
+/*
```

Comment:
General comment.  Instead of StringValue we should use "optional string field = 1".  Proto3 was revised to support this [for checking for presence](https://stackoverflow.com/questions/42622015/how-to-define-an-optional-field-in-protobuf-3)

Its not clear to me if optional should be used in all cases (it might be sufficient to denote empty as interpreted as not specified).

Reply:
Thank you so much for your review @emkornfield !

We tried to use `optional` before, but it is not available in the protobuf version the project is using (3.7.1), I wonder if it's better to keep it using StringValue or update the library?

Reply:
My preference would be to update the library.  This might be painful on the C++ side but I think it is the right thing to do in the long term.  @lidavidm thoughts?

Reply:
I would be a little worried about supporting LTS releases of Ubuntu and such. It does seem to be in conda-forge already, at least.

And we already vary our Protobuf version based on features requested: https://github.com/apache/arrow/blob/04575bc8e90216d8e685d1b857f487708426001c/cpp/cmake_modules/ThirdpartyToolchain.cmake#L1415-L1424 so I suppose this wouldn't change the situation there (20.04 LTS ships Protobuf 3.6 anyways).

We may also have to bump Protobuf if/when we upgrade gRPC, though, I cannot figure out what exactly gRPC requires even after some searching. It seems v1.38 uses Protobuf 3.15 though it is not clear if that is a hard floor or not. 

@kszucs how much trouble would it be to bump our Protobuf version?

Reply:
@kou can probably tell. I'd say that we should try to bump it, trigger the nightly builds and check where can we switch to bundled builds instead of using system protobuf.

Reply:
There is no problem we require Protobuf 3.15 or later only when Apache Arrow Flight is enabled.
We just need to bump bundled Protobuf version for it.

Reply:
https://github.com/apache/arrow/pull/11006 was just merged thanks to Kou. You should be able to use the Protobuf optionals now on the C++ side (you should be able to bump the Java version easily enough?)